lib/sarif/npm_audit_sarif.rb

```diff
@@ -0,0 +1,49 @@
+module Sarif
+  class NPMAuditSarif < BaseSarif
+    NPM_URI = 'https://docs.npmjs.com/cli/v7/commands/npm-audit'.freeze
+
+    def initialize(scan_report)
+      super(scan_report)
+      @uri = NPM_URI
+      @logs = @scan_report.to_h[:info][:stdout][:advisories].values
+      @issues = Set.new
+    end
+
+    def parse_issue(issue)
+      id = issue[:id].to_i
+      return nil if @issues.include?(id)
+
+      @issues.add(id)
+      {
+        id: format('NPM%<number>.4d', number: issue[:id].to_i),
```

Comment:
Can the id be like `39` instead of `NPM0039`?   Would sarif schema allow the former?

Reply:
it doesnt really matter, I'm only following the format that was used in the [SARIF doc](https://docs.oasis-open.org/sarif/sarif/v2.1.0/sarif-v2.1.0.pdf) (RuleID: 3.27.5)

Reply:
its not a schema violation, I only used that for consistency. If you want it changed, i can update it

Reply:
@joshua-ostrom  Do you think the id should look like `39` which matches the id in the NPM report?